### DISCUSSION TITLE
Select partial column is broken.

### DISCUSSION THREAD

**Discussion by kev0960** [SeaQL/sea-orm#459] [Q&A] [answered]
I was trying to select only a couple of column using

```
Data::Entity().find().select_only(Data::Column::SomeColumn).all(&db).await?
```

But this fails saying some other column in Data is missing.

My guess is that because what all() returns is an Entity, it needs to have all columns in Entity to be returned. This will be impossible for this case since we are only fetching partial columns.

Is there a way to make all to return not an Entity but partially set values? like ActiveModel? (where some are set and some are not)

**Comment by billy1624** [comment]
Hey @kev0960, you have to select the partial result into a custom struct.

https://www.sea-ql.org/SeaORM/docs/advanced-query/custom-select#handling-custom-selects

**Comment by billy1624** [comment, marked as answer]
For example...

```rust
#[derive(FromQueryResult)]
struct CustomModel {
    some_column: String,
}

let custome_model: Vec<CustomModel> = Data::Entity()
      .find()
      .select_only(Data::Column::SomeColumn)
      .all(&db)
      .into_model()
      .await?
```

**Comment by kev0960** [comment]
Oh Nice. Thank you :)
